File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/module/ModuleLogger.java
package de.jalumu.magma.platform.base.module;

import java.util.logging.Logger;

public class ModuleLogger extends Logger {

    protected ModuleLogger(String moduleName, Logger parent) {
        super("MagmaKT-" + moduleName, null);
        setParent(parent);
        setLevel(parent.getLevel());
    }
}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/module/ModuleLogger.kt
package de.jalumu.magma.platform.base.module

import java.util.logging.Logger

class ModuleLogger constructor(moduleName: String, parent: Logger) : Logger(
    "MagmaKT-$moduleName", null
) {
    init {
        setParent(parent)
        level = parent.level
    }
}
